fe/fe-core/src/test/java/org/apache/doris/nereids/trees/expressions/ExpressionParserTest.java

```diff
@@ -137,4 +137,13 @@ public void testSortClause() throws Exception {
         String sort1 = "select a from test order by 1";
         assertSql(sort1);
     }
+
+    @Test
+    public void testCaseWhen() throws Exception {
+        String caseWhen = "select case a when 1 then 2 else 3 end from test";
+        assertSql(caseWhen);
+
+        String caseWhen2 = "select case when a = 1 then 2 else 3 end from test";
```

Comment:
I think should another add some ut case. add complex expr after then and else?

Reply:
@yinzhijian add ut later